code/modules/uplink/uplink_item.dm

```diff
@@ -110,10 +110,10 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
 	var/obj/item/weapon/storage/box/B = A
 	if(istype(B) && B.contents.len > 0)
 		for(var/obj/item/I in B)
-			U.purchase_log += "<big>[bicon(I)]</big>"
+			U.purchase_log += "<big>[icon2base64html(I)]</big>"
 	else
 		if(purchase_log_vis)
-			U.purchase_log += "<big>[bicon(A)]</big>"
+			U.purchase_log += "<big>[icon2base64html(A)]</big>"
```

Comment:
we can, but for this, the log is sent to all people, but at the end of the round, how ever, the log to be sent is generated during the round.

The issue here is the string is stored, and sent much much later, so we can't actually know at generation time who needs to see it.

This is where bicon is ok.